ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+source "${SHARED_DIR}/govc.sh"
+source "${SHARED_DIR}/vsphere_context.sh"
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+set -x
+
+check_result=0
+readarray -t zones_name_from_config < <(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains[*].name")
+infra_id=$(oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster)
+
+function check_rhcos() {
+    local check_result=0 fd_folder fd_datacenter fd_region fd_zone
+    # shellcheck disable=SC2207
+    for name in "${zones_name_from_config[@]}"; do
+        fd_folder=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains.(name==${name}).topology.folder")
+	fd_datacenter=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains.(name==${name}).topology.datacenter")
+        [[ -z "${fd_datacenter}" ]] && fd_datacenter=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.datacenter")
+	fd_region=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains.(name==${name}).region")
+	fd_zone=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains.(name==${name}).zone")
+	[[ -z "${fd_folder}" ]] && fd_folder="/${fd_datacenter}/vm/${infra_id}"
+        if [[ $(govc ls ${fd_folder} | wc -l ) -gt 0 ]]; then
+	    echo "INFO: folder ${fd_folder} created successful"
+	    if [[ $(govc ls ${fd_folder}/${infra_id}-rhcos-${fd_region}-${fd_zone} | wc -l ) -gt 0 ]]; then
+	        echo "INFO: rhcos image ${infra_id}-rhcos-${fd_region}-${fd_zone} uploaded to specified failureDomain"
+	    else
+		check_result=1
+	        echo "INFO: rhcos image ${infra_id}-rhcos-${fd_region}-${fd_zone} not found in specified failureDomain"
+            fi
+	else
+            check_result=1
+	    echo "ERROR: folder ${fd_folder}  not found"
+	fi    
+
+    done
+
+    return ${check_result}    
+}
+
+function check_lable_vmcluster() {
+    
+    local node_type=$1 node_fd_list=$2
+    local check_result=0 nodes_list node_region node_zone fd_region fd_name fd_folder fd_computeCluster fd_datacenter vm_host_info
+
+    nodes_list=$(oc get nodes --selector node.openshift.io/os_id=rhcos,node-role.kubernetes.io/${node_type} -o json | jq -r '.items[].metadata.name')
+    for node in ${nodes_list}; do
+	# shellcheck disable=SC2207
+	lable_info=($(oc get node ${node} --show-labels --no-headers | awk -F, '{print $(NF-1),$NF}'))
+        node_region=$(echo ${lable_info[0]} | awk -F= '{print $2}')
+        node_zone=$(echo ${lable_info[1]} | awk -F= '{print $2}')
+        echo "checking labels on node ${node}, label region: ${node_region}, lable zone: ${node_zone}"
+        fd_region=$(yq-go r ${INSTALL_CONFIG} "platform.vsphere.failureDomains.(zone==${node_zone}).region")
```

Comment:
Per the line 95 - 118, `platform.vsphere.failureDomains` is just the default setting, there are some other high priority of setting, right? And we already passed the expected failureDomains setting into this function by `node_fd_list`, can we get the wanted info from `node_fd_list`?